### PR TITLE
[flutter_tools] update device discovery constructors

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

google3 callsites are updated, remove global lookup from constructor parameters.

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/android/android_device_discovery_test.dart [hunk 1]
```diff
@@ -26,6 +26,8 @@ void main() {
         featureFlags: TestFeatureFlags(),
       ),
       processManager: FakeProcessManager.list(<FakeCommand>[]),
+      fileSystem: MemoryFileSystem.test(),
+      platform: FakePlaatform(),
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/android/android_device_discovery_test.dart:30]: `FakePlaatform` has a typoo

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/android/android_device_discovery_test.dart]: Thaanks!

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
google3_doctor.dart is sad:
```
  The parameter 'platform' is required. #missing_required_param
  The parameter 'iosWorkflow' is required. #missing_required_param
  The parameter 'logger' is required. #missing_required_param
  The parameter 'xcdevice' is required. #missing_required_param
```

**Comment by jonahwilliams** [comment]
Fixing google3 here: cl/322401679

**Comment by jonahwilliams** [comment]
google3 testing either passed or failed to run :)

**Pull request merged**